addon/components/o-s-s/currency-input.ts

```diff
@@ -111,6 +122,23 @@ export default class OSSCurrencyInput extends Component<OSSCurrencyInputArgs> {
     this.currencySelectorShown = false;
     this.filteredCurrencies = this._currencies;
   }
+
+  private _handlePaste(event: ClipboardEvent): void {
+    event.preventDefault();
+
+    let paste = event.clipboardData?.getData('text') || '';
+    paste = paste.replace(NOT_NUMERIC_FLOAT, '');
+
+    const target = event.target as HTMLInputElement;
+    const initialSelectionStart = target.selectionStart || 0;
+    const finalSelectionPosition = initialSelectionStart + paste.length;
+
+    target.setRangeText(paste, initialSelectionStart, target.selectionEnd || initialSelectionStart);
+    target.setSelectionRange(finalSelectionPosition, finalSelectionPosition);
+
+    this.localValue = target.value as unknown as number;
```

Comment:
Use `as unknown as number` instead of `Number(str)` to avoid scientific notation transformation in input